### DISCUSSION TITLE
One or multiple buckets for Thanos Sidecar of HA Prometheus?

### DISCUSSION THREAD

**Discussion by dastrobu** [thanos-io/thanos#4378] [Questions & Answers] [answered]
I am currently working on a HA Prometheus setup with >= 2 replicas and a Thanos Sidecar on each Prometheus replica. 

The question that came up (and I did not find the answers in the docs) is if I have to use multiple buckets (one for each replica) or if I can safely use the same bucket on each Thanos Sidecar. 
Is is safe to let multiple Sidecars upload to the same bucket? 
Would that be good or bad practice? 

Thanks for some thoughts on this or just pointing me at the right spot in the docs.

**Comment by fktkrt** [comment, marked as answer]
Hey, you can use a single object store for multiple replicas, in fact, I'd say that using a single store for a cluster can be considered best practice. 

Note that you can use a single object store for multiple clusters as well (I think Prometheus will have unique IDs for all the blocks, but), optimally, you would need unique external labels configured per sidecars/replicas to have deduplication and easy differentiation across replicas. 

If you'd later go multi-cluster, one thing you should pay attention to is that a single object store comes with a single compactor and a StoreGateway.
